daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetImpl.scala

```diff
@@ -1252,15 +1260,15 @@ sealed class DISimple(override val erd: ElementRuntimeData)
           else if (f == Float.NegativeInfinity) XMLUtils.NegativeInfinityString
           else f.toString
         }
-        case _ => dataValue.toString
+        case x => x.toString
```

Comment:
Should we make dataValue.toString cause an Assert.usageError? Or is messing with toString a super bad idea (e.g., blows up the debugger???)

Reply:
if I call toString on a scala Int, you're saying they box the Int then, and then invoke toString on it?  Or do they treat built in types differently from user defined AnyVal types?

Reply:
Just tested this:

`object Test{
  def main(args: Array[String]): Unit = { 
    val x:Int = 1;
    val y=x.toString()
  }
}
`

Compiles to:

`
public final class Test$ {
  public static final Test$ MODULE$;

  public static {};
    Code:
       0: new           #2                  // class Test$
       3: invokespecial #12                 // Method "<init>":()V
       6: return

  public void main(java.lang.String[]);
    Code:
       0: iconst_1
       1: istore_2
       2: iload_2
       3: invokestatic  #20                 // Method scala/runtime/BoxesRunTime.boxToInteger:(I)Ljava/lang/Integer;
       6: invokevirtual #24                 // Method java/lang/Object.toString:()Ljava/lang/String;
       9: astore_3
      10: return
`

Switching to
`

Reply:
Yeah, I guess since toString is a virtual function, you must have an object to invoke it on. 

Hopefully for some set of small ints, they return the same object rather than allocating new ones.